server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -44,15 +44,21 @@ private GetHealthAction() {
         private final ClusterName clusterName;
         private final HealthStatus status;
         private final List<HealthComponentResult> components;
+        private final boolean showTopLevelStatus;
 
         public Response(StreamInput in) {
             throw new AssertionError("GetHealthAction should not be sent over the wire.");
         }
 
-        public Response(final ClusterName clusterName, final List<HealthComponentResult> components) {
-            this.clusterName = clusterName;
+        public Response(final ClusterName clusterName, final List<HealthComponentResult> components, boolean showTopLevelStatus) {
```

Comment:
I think the REST side is a bit too smart here. If a component is requested it should always compute the status IMO.

I wonder if this should have another constructor that only takes one `HealthIndicatorResult` when we drilled all the way down to one indicator? This way each constructor will do populate the necessary fields based on the use-case?

This could possibly remove the need for the other `showComponentSummary` flag in `HealthComponentResult` https://github.com/elastic/elasticsearch/pull/85234/files#diff-9b325d699eee9dac56052837a367c89ec0b4613a2a28c0c4cc44e59ae6b81f99R19

What do you think?

Reply:
Isn't this analogous to the top-level request? If you drill down to a component, we don't generate the status of the higher level thing (the status of the component is all that's relevant at that point). If you drill down to an indicator, we don't generate the status of the higher level thing (the status of the indicator is all that's relevant at that point).